utils/swipe.js

```diff
@@ -0,0 +1,109 @@
+export default class Swipe {
+
+  constructor(elm) {
+    this.handleMove = this.handleMove.bind(this);
+    this.handleStart = this.handleStart.bind(this);
+    this.handleEnd = this.handleEnd.bind(this);
+
+    this.$elm = typeof elm === 'string' ? document.querySelector(elm) : elm;
+    this.hasTouch = 'ontouchstart' in window;
+    this.bindEvents();
+  }
+
+  bindEvents() {
+    if (this.hasTouch) {
+      this.$elm.addEventListener('touchstart', this.handleStart, false);
+      this.$elm.addEventListener('touchmove', this.handleMove, false);
+      this.$elm.addEventListener('touchend', this.handleEnd, false);
+    } else {
+      this.$elm.addEventListener('mousedown', this.handleStart, false);
+      this.$elm.addEventListener('mousemove', this.handleMove, false);
+      this.$elm.addEventListener('mouseup', this.handleEnd, false);
+    }
+  }
+
+  unbind() {
+    if (this.hasTouch) {
+      this.$elm.removeEventListener('touchstart', this.handleStart, false);
+      this.$elm.removeEventListener('touchmove', this.handleMove, false);
+      this.$elm.removeEventListener('touchend', this.handleEnd, false);
+    } else {
+      this.$elm.removeEventListener('mousedown', this.handleStart, false);
+      this.$elm.removeEventListener('mousemove', this.handleMove, false);
+      this.$elm.removeEventListener('mouseup', this.handleEnd, false);
+    }
+  }
+
+  onLeft(callback) {
```

Comment:
This style of API is a bit strange, since it doesn't allow you to redefine a callback handler, since it redefines itself, which is not obvious, can you consider the following change?

```
onLeft(callback) {
  this.onLeftHandler = callback;
  return this;
}
```